compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -3693,4 +3702,26 @@ private void loadStructField(BLangExpression fieldAccessExpr, Operand varRefRegI
             emit(opcode, targetRegIndex, calcAndGetExprRegIndex(fieldAccessExpr));
         }
     }
+
+    private void setVariableScopeStart(LocalVariableInfo localVarInfo, BLangVariable varNode) {
+        if (varNode.pos != null) {
+            localVarInfo.scopeStartLineNumber = varNode.pos.sLine;
+        }
+    }
+
+    private void setVariableScopeEnd(LocalVariableInfo localVarInfo, BLangVariable varNode) {
+        if ((varNode.parent == null) && (varNode.pos != null)) {
+            localVarInfo.scopeEndLineNumber = varNode.pos.eLine;
+        } else {
```

Comment:
shall we return here and improve the else part?

Reply:
Ack; Modified as suggested.